### PR TITLE
Add imagePullSecrets in kube-agent chart

### PR DESCRIPTION
**Pull request by amemni**
The `imagePullSecrets` attribute would be useful when pulling images from a Docker proxy repository/private registry which is something very common now, especially for caching Docker images and working around the Dockerhub download rate limit.

Changes:
* Modified [teleport-kube-agent/templates/config.yaml](https://github.com/highb/teleport/blob/master/examples/chart/teleport-kube-agent/templates/deployment.yaml) to set the `imagePullSecrets` if defined,
* Documented it in [teleport-kube-agent/README.md](https://github.com/highb/teleport/blob/master/examples/chart/teleport-kube-agent/README.md)

### CODE REVIEW

**Review by aletournel — APPROVED** [review]
LGTM

**Comment by russjones** [comment]
@awly @r0mant Can you review?

File: examples/chart/teleport-kube-agent/templates/deployment.yaml [hunk 1]
```diff
@@ -65,6 +65,10 @@ spec:
     {{- toYaml .Values.extraVolumeMounts | nindent 8 }}
   {{- end }}
 {{- end }}
+      {{- if .Values.imagePullSecret }}
+      imagePullSecrets:
+      - name: {{ .Values.imagePullSecret }}
+      {{- end }}
```
Comment (awly) [inline examples/chart/teleport-kube-agent/templates/deployment.yaml:71]: nit: let's move this above `initContainers`

File: examples/chart/teleport-kube-agent/values.yaml [hunk 2]
```diff
@@ -64,6 +64,8 @@ labels: {}
 # Container image for the agent. Since this runs without the auth_service, we
 # don't need the enterprise version.
 image: quay.io/gravitational/teleport
+# Optional secret to use when pulling image from a private registry
+imagePullSecret: ""
```
Comment (awly) [inline examples/chart/teleport-kube-agent/values.yaml:68]: Could you restructure this similar to https://github.com/gravitational/teleport/blob/ab02b4dad8aea8a92c34820bbe254e0c01073e56/examples/chart/teleport/values.yaml#L20-L21 ?

**Review by awly — COMMENTED** [review]
@webvictim PTAL

File: examples/chart/teleport-kube-agent/templates/deployment.yaml [hunk 3]
```diff
@@ -65,6 +65,10 @@ spec:
     {{- toYaml .Values.extraVolumeMounts | nindent 8 }}
   {{- end }}
 {{- end }}
+      {{- if .Values.imagePullSecret }}
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/templates/deployment.yaml:68]:
```suggestion
      {{- if .Values.imagePullSecrets }}
```

File: examples/chart/teleport-kube-agent/templates/deployment.yaml [hunk 4]
```diff
@@ -65,6 +65,10 @@ spec:
...
+      imagePullSecrets:
+      - name: {{ .Values.imagePullSecret }}
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/templates/deployment.yaml:70]:
```suggestion
      - name: {{ .Values.imagePullSecrets }}
```

File: examples/chart/teleport-kube-agent/templates/deployment.yaml (hunk 1 above)
Comment (amemni) [reply examples/chart/teleport-kube-agent/templates/deployment.yaml]: fix: https://github.com/gravitational/teleport/pull/6941/commits/8e11fb6f669e3dc6e25e89436efc874c1146999e

File: examples/chart/teleport-kube-agent/values.yaml (hunk 2 above)
Comment (amemni) [reply examples/chart/teleport-kube-agent/values.yaml]: fix: https://github.com/gravitational/teleport/pull/6941/commits/8e11fb6f669e3dc6e25e89436efc874c1146999e

File: examples/chart/teleport-kube-agent/templates/deployment.yaml (hunk 3 above)
Comment (amemni) [reply examples/chart/teleport-kube-agent/templates/deployment.yaml]: fix: https://github.com/gravitational/teleport/pull/6941/commits/8e11fb6f669e3dc6e25e89436efc874c1146999e

File: examples/chart/teleport-kube-agent/templates/deployment.yaml (hunk 4 above)
Comment (amemni) [reply examples/chart/teleport-kube-agent/templates/deployment.yaml]: fix: https://github.com/gravitational/teleport/pull/6941/commits/8e11fb6f669e3dc6e25e89436efc874c1146999e

**Review by awly — APPROVED** [review]

**Comment by russjones** [comment]
@webvictim Can you review, @awly has approved.

**Comment by webvictim** [comment]
@amemni Thanks for the PR. Please also:
- add a section for `imagePullSecrets` to the appropriate section of the [`teleport-kube-agent` Helm chart reference](https://github.com/gravitational/teleport/blob/master/docs/pages/kubernetes-access/helm/reference.mdx)
  - see the [`initContainers`](https://goteleport.com/docs/kubernetes-access/helm/reference/#initcontainers-1) section for an example.
- add a file (`examples/chart/teleport-kube-agent/.lint/imagepullsecrets.yaml`) with an example value for the linter - something like this:
```yaml
authToken: auth-token
proxyAddr: proxy.example.com:3080
roles: kube
kubeClusterName: test-kube-cluster
imagePullSecrets:
- name: myRegistryCredentials
```
- add `imagePullSecrets` at the correct location in [`examples/chart/teleport-kube-agent/values.schema.json`](https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-kube-agent/values.schema.json) so that it can be validated by Helm:
```json
"imagePullSecrets": {
    "$id": "#/properties/imagePullSecrets",
    "type": "array",
    "default": []
},
```

Thanks.

**Comment by webvictim** [comment]
```
==> Linting examples/chart/teleport
--
144 |  
145 | 1 chart(s) linted, 0 chart(s) failed
146 | WARNING: This chart is deprecated
147 | Using values from 'examples/chart/teleport-kube-agent/.lint/affinity.yaml': ==> Linting examples/chart/teleport-kube-agent
148 | [ERROR] values.yaml: - imagePullSecrets: Invalid type. Expected: array, given: null
```

File: examples/chart/teleport-kube-agent/values.yaml [hunk 5]
```diff
@@ -84,6 +84,9 @@ storage:
 # Container image for the agent. Since this runs without the auth_service, we
 # don't need the enterprise version.
 image: quay.io/gravitational/teleport
+# Optional array of imagePullSecrets, to use when pulling from a private registry
+imagePullSecrets:
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/values.yaml:88]:
```suggestion
imagePullSecrets: []
```

**Review by webvictim — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Review by webvictim — CHANGES_REQUESTED** [review]
@amemni It looks like the linter is still failing the tests on this PR:

```
Using values from 'examples/chart/teleport-kube-agent/.lint/imagepullsecrets.yaml': examples/chart/teleport-kube-agent/.lint/imagepullsecrets.yaml
--
169 | 1:22      error    wrong new line character: expected \n  (new-lines)
170 |  
171 | 1	authToken: auth-token^M$
172 | 2	proxyAddr: proxy.example.com:3080^M$
173 | 3	roles: kube^M$
174 | 4	kubeClusterName: test-kube-cluster^M$
175 | 5	image: quay.io/gravitational/teleport^M$
176 | 6	imagePullSecrets:^M$
177 | 7	- name: myRegistryKeySecretName^M$
178 | Makefile:451: recipe for target 'lint-helm' failed
179 | make: *** [lint-helm] Error 1
```

I think the reason is that you're using an editor that's writing newlines as `\r` rather than `\n`. Please make sure all files you've changed are being written with UNIX line endings.

**Review by webvictim — APPROVED** [review]
LGTM, thanks!

**Pull request merged**